app/views/saved_scenarios/reports/show.csv.erb

```diff
@@ -0,0 +1,8 @@
+<%=@scenario_id%>
+<%-@yml.each do |category, value| %>
+  <%-value.each do |subcategory, subvalue| %>
+    <%-subvalue.each do |split, query| %>
+      <%=category%>;<%=subcategory%>;<%=split%>;<%=@query_api_response[query]['future']%>;<%=@query_api_response[query]['unit']%>
```

Comment:
Being a bit nitpicky, but the CSVs that ETEngine generates use commas as separators rather than semicolons. The indenting of this line also creates extra leading whitespace in the CSV file; this might mess up some tools which don't ignore it.

```
1214450
      biomassa;;collectief;5604795553.98533;MJ
      biomassa;;individueel;21958769999.989735;MJ
      biomassa;;transport;12451999999.999996;MJ
      biomassa;;industrie;5913489241.570863;MJ
# ...
```

```suggestion
<%=category%>;<%=subcategory%>;<%=split%>;<%=@query_api_response[query]['future']%>;<%=@query_api_response[query]['unit']%>
```

Perhaps it would also be nice to add a header row (instead of the scenario ID?) to describe each column?

```csv
Category,Subcategory,Split,Value,Unit
```